src/core/cgroup.h

```diff
@@ -144,17 +144,29 @@ struct CGroupContext {
 
         uint64_t default_memory_min;
         uint64_t default_memory_low;
+        uint64_t default_startup_memory_low;
         uint64_t memory_min;
         uint64_t memory_low;
+        uint64_t startup_memory_low;
         uint64_t memory_high;
+        uint64_t startup_memory_high;
         uint64_t memory_max;
+        uint64_t startup_memory_max;
         uint64_t memory_swap_max;
+        uint64_t startup_memory_swap_max;
         uint64_t memory_zswap_max;
+        uint64_t startup_memory_zswap_max;
```

Comment:
i don't really like the proliferation of so many fields that basically carry the same data.

we probably should rework this so that this becomes:

```
typedef enum CGroupMemoryLimitType {
        _CGROUP_MEMORY_LIMIT_DEFAULT_MIN,
        _CGROUP_MEMORY_LIMIT_DEFAULT_LOW,
        …
```

and then add an array here:

```c
…
uint64_t memory_limits[_CGROUP_MEMORY_LIMIT_MAX];
…
```